include/deal.II/grid/reference_cell.h

```diff
@@ -183,6 +183,77 @@ namespace ReferenceCell
     return Point<dim>(+0.0, +0.0, +0.0);
   }
 
+  /**
+   * Return i-th unit tangential vector (array of dim-1 elements of Point<dim>)
+   * of a face of the reference cell. The vectors are arranged such that the
+   * cross product between the two vectors return the unit normal vector.
+   */
+  template <int dim>
+  inline Tensor<1, dim>
+  unit_tangential_vectors(const Type &       reference_cell,
+                          const unsigned int face_no,
+                          const unsigned int i)
+  {
+    if (reference_cell == get_hypercube(dim))
+      return GeometryInfo<dim>::unit_tangential_vectors[face_no][i];
+    else if (reference_cell == Type::Tri)
+      {
+        AssertDimension(i, 0);
+        static const std::array<Tensor<1, dim>, 3> table = {
```

Comment:
Maybe make it `constexpr`? I wonder if the whole function could actually be `constexpr`.

That's also not important: If it's easy to do, do it. Otherwise leave it for someone to clean up later.

Reply:
This is not possible since there is an conversion from `Point` to `Tensor`. The reason why `Points` is used here (and I guess at other places from where I had the idea write it in such a way) it that `Point` has appropriate constructors taking 1-3 doubles. The last point rules out that we can simplify the code with `{}` as proposed by you below.

Reply:
But class `Tensor` has this constructor:
```
  using array_type =
    typename Tensor<rank_ - 1, dim, Number>::array_type[(dim != 0) ? dim : 1];

  constexpr DEAL_II_CUDA_HOST_DEV explicit Tensor(
    const array_type &initializer);
```
You might have to use an extra pair of braces, but you can direct-initialize the `Tensor` objects without the detour of `Point`.

Reply:
That is an explicit constructor. So it will not work.